src/encoded/static/components/app.js

```diff
@@ -608,25 +608,43 @@ export default class App extends React.Component {
         // Skip @@download links
         if (href.indexOf('/@@download') != -1) return;
 
-        // Don't cache requests to user profile.
-        var navOpts = {};
-        if (target.getAttribute('data-no-cache')) navOpts.cache = false;
-
         // With HTML5 history supported, local navigation is passed
         // through the navigate method.
         if (this.historyEnabled) {
             event.preventDefault();
-            navigate(href, navOpts, ()=>{
-                var hrefParts = url.parse(href);
-                var hrefHash = hrefParts.hash;
+
+            var navOpts      = {},
+                hrefParts    = url.parse(href),
+                pHrefParts   = url.parse(this.props.href),
+                hrefHash     = hrefParts.hash,
+                targetOffset = target.getAttribute('data-target-offset'),
+                noCache      = target.getAttribute('data-no-cache');
```

Comment:
This change allows link elements to have attributes that help inform navigation. E.g. `data-target-offset` would be how far above the target element to scroll to if link href contains a target element ID, e.g. `http://someurl.com/a/page/somewhere#id`. More options can be added as needed, including overrides for navigation options (`navOpts`).